toolkit/tools/imagecustomizerapi/module.go

```diff
@@ -5,37 +5,61 @@ package imagecustomizerapi
 
 import (
 	"fmt"
+	"strings"
 )
 
+type ModuleOptions map[string]string
+
 type Module struct {
-	Name string `yaml:"Name"`
+	Name     string            `yaml:"Name"`
+	LoadMode string            `yaml:"LoadMode"`
+	Options  map[string]string `yaml:"Options"`
 }
 
 func (m *Module) IsValid() error {
-	if m.Name == "" {
-		return fmt.Errorf("name of module may not be empty")
+	if err := validateModuleName(m.Name); err != nil {
+		return err
+	}
+
+	if err := validateModuleLoadMode(m.LoadMode); err != nil {
+		return err
 	}
 
+	if len(m.Options) > 0 {
+		for optionKey, optionValue := range m.Options {
+			if optionKey == "" || optionValue == "" {
+				return fmt.Errorf("option key or value cannot be empty for module %s", m.Name)
```

Comment:
This error message could be misinterpreted as saying that either the key or the value must not be empty. That is, they can't both be empty at the same time. It might be better to give the key and the value their own error message.

Same thing for the spaces/newlines error message below.